openwisp_network_topology/api/views.py

```diff
@@ -39,46 +47,54 @@ class RequireAuthentication(APIView):
         ]
 
 
-class NetworkCollectionView(generics.ListAPIView, RequireAuthentication):
+class ListViewPagination(pagination.PageNumberPagination):
+    page_size = 10
+    page_size_query_param = 'page_size'
+    max_page_size = 100
+
+
+class NetworkCollectionView(
+    RequireAuthentication, FilterByOrganizationManaged, generics.ListCreateAPIView,
+):
     """
     Data of all the topologies returned
-    in NetJSON NetworkCollection format
+    in NetJSON NetworkCollection format.
     """
 
     serializer_class = NetworkGraphSerializer
-    queryset = Topology.objects.filter(published=True)
+    queryset = Topology.objects.select_related('organization')
```

Comment:
Somethings are confusing, why do we not need published filter now?

Reply:
I tried to keep it like the admin panel. If not required should I undo it??

Reply:
Let's ask @nemesisdesign why there was publish filter only in API before, because of change management, we might not want to change it! :smile:

Reply:
@nemesisdesign ping!